src/asciidocParser.ts

```diff
@@ -2,26 +2,21 @@ import * as vscode from 'vscode'
 import * as path from 'path'
 import { spawn } from 'child_process'
 
-const asciidoctor = require('@asciidoctor/core')()
+const asciidoctor = require('@asciidoctor/core')
 const docbook = require('@asciidoctor/docbook-converter')
 const kroki = require('asciidoctor-kroki')
-const highlightjsBuiltInSyntaxHighlighter = asciidoctor.SyntaxHighlighter.for('highlight.js')
+const highlightjsBuiltInSyntaxHighlighter = asciidoctor().SyntaxHighlighter.for('highlight.js')
 const highlightjsAdapter = require('./highlightjs-adapter')
 
-const useKroki = vscode.workspace.getConfiguration('asciidoc', null).get('use_kroki')
-if (useKroki) {
-  kroki.register(asciidoctor.Extensions)
-}
-
 export class AsciidocParser {
   public html: string = ''
   public document = null
+  public processor = null
   private extPath = vscode.extensions.getExtension('asciidoctor.asciidoctor-vscode').extensionPath
   private stylesdir = path.join(this.extPath, 'media')
 
   constructor (private readonly filename: string, private errorCollection: vscode.DiagnosticCollection = null) {
-    this.filename = filename
-    this.errorCollection = errorCollection
+    this.processor = asciidoctor()
```

Comment:
I don't think we need to instantiate the `processor` in the constructor, do we?
Most of the time we can do:

```js
const processor = require('@asciidoctor/core')()
```

In the future, we might be able to write:

```js
import { convert } from '@asciidoctor/core'
```

The reason we have a constructor in Asciidoctor.js is merely a workaround because Opal will modify basic object prototypes. So in some cases, it's important to decide when Opal is effectively loaded.

```js
// Opal not yet loaded
const Asciidoctor = require('@asciidoctor/core')

// ... do some work

// Load Opal!
const processor = Asciidoctor()
```

Reply:
I have a habit of "objectifying" everything.... I will fix.